File: python/scripts/bump_version.py
import semver
import toml

# Define your pyproject.toml and version.py file paths
pyprojectTomlPath = "pyproject.toml"
versionPyPath = "retoolrpc/version.py"

# Read the current pyproject.toml
with open(pyprojectTomlPath, "r") as tomlFile:
    pyprojectToml = toml.load(tomlFile)

# Read the current version from version.py
currentVersion = pyprojectToml["tool"]["poetry"]["version"]
# You can use 'bump_minor' or 'bump_major' to increment accordingly
newVersion = semver.bump_patch(currentVersion)

# Update version in pyproject.toml
pyprojectToml["tool"]["poetry"]["version"] = newVersion
with open(pyprojectTomlPath, "w") as tomlFile:
    toml.dump(pyprojectToml, tomlFile)

# Update version in version.py
versionPyContent = f'__version__ = "{newVersion}"\n'
with open(versionPyPath, "w") as versionPyFile:
    versionPyFile.write(versionPyContent)

print(f"Version bumped to {newVersion}")

File: javascript/scripts/bumpVersion.ts
import fs from 'fs-extra'
import semver from 'semver'

// Define your package.json and version.ts file paths
const packageJsonPath = './package.json'
const versionTsPath = './src/version.ts'

// Read the current package.json
const packageJson = fs.readJsonSync(packageJsonPath)

// Increment the version
const newVersion = semver.inc(packageJson.version, 'patch') // You can use 'minor' or 'major' instead of 'patch' to increment accordingly

// Update package.json version
packageJson.version = newVersion
fs.writeJsonSync(packageJsonPath, packageJson, { spaces: 2 })

// Update version.ts file
const versionTsContent = `export const RetoolRPCVersion = '${newVersion}'\n`
fs.writeFileSync(versionTsPath, versionTsContent)

console.log(`Version bumped to ${newVersion}.`)
